telegram/ext/_application.py

```diff
@@ -1019,14 +1035,15 @@ def __create_task_done_callback(self, task: asyncio.Task) -> None:
 
     async def __create_task_callback(
         self,
-        coroutine: Union[Generator[Optional["asyncio.Future[object]"], None, RT], Awaitable[RT]],
+        coroutine: _CoroType,
         update: Optional[object] = None,
         is_error_handler: bool = False,
     ) -> RT:
         try:
-            if isinstance(coroutine, Generator):
+            # Generator-based coroutines are not supported in Python 3.12+
+            if isinstance(coroutine, Generator) and sys.version_info < (3, 12):
                 return await asyncio.create_task(coroutine)
-            return await coroutine
+            return await coroutine  # type: ignore[misc]
```

Comment:
Because I had to ignore this here. 
`Incompatible types in "await" (actual type "Union[Generator[Optional[Future[object]], None, Any], Awaitable[Any]]", expected type "Awaitable[Any]")  [misc]
`